pkg/northbound/admin/admin.go

```diff
@@ -54,6 +57,64 @@ func (s Server) RegisterModel(ctx context.Context, req *RegisterRequest) (*Regis
 	}, nil
 }
 
+// UploadRegisterModel uploads and registers a new model plugin.
+func (s Server) UploadRegisterModel(stream ConfigAdminService_UploadRegisterModelServer) error {
+	response := RegisterResponse{Name: "Unknown"}
+	soFileName := ""
+
+	const TEMPFILE = "/tmp/uploaded_model_plugin.tmp"
+	f, err := os.Create(TEMPFILE)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create temporary file %s", TEMPFILE)
+	}
+	defer f.Close()
+
+	// while there are messages coming
+	i := 0
+	for {
+		chunk, err := stream.Recv()
+		if err != nil {
+			if err == io.EOF {
+				goto END
```

Comment:
would break be enough here ? I don't see the point fo the goto

Reply:
I copied this from an example on the web - I'll use break instead